pyomo/core/expr/numvalue.py

```diff
@@ -223,11 +216,7 @@ def is_constant(obj):
     #
     if obj.__class__ in native_types:
         return True
-    try:
-        return obj.is_constant()
-    except AttributeError:
-        pass
-    return as_numeric(obj).is_constant()
+    return obj.is_constant()
```

Comment:
Again, as with value, there is no guarantee that the first time Pyomo sees a new class object is in the expression system.  It could be through any of the "global scope" information methods: `value()`, `is_constant()`, etc...

The advantage of the:
```
try:
    return obj.is_constant()
except AttributeError:
    return as_numeric(obj).is_constant()
```
idiom is that it costs NO time in the nominal case, but works correctly in the exceptional case where this is the first time that Pyomo sees a new class.